src/geotiffwriter.js

```diff
@@ -252,11 +252,12 @@ const encodeImage = (values, width, height, metadata) => {
 
   const prfx = new Uint8Array(encodeIfds([ifd]));
 
-  const img = new Uint8Array(values);
+  // This should allow other kinds of typed arrays to be used. Uint8 is just a view of the real underlying data.
+  const img = new Uint8Array(values.buffer);
```

Comment:
Unfortunately, values is sometimes a simple JavaScript array and not a buffer, so there is no `.buffer` property available in that case.  Here's where this simple array is created: https://github.com/geotiffjs/geotiff.js/blob/master/src/geotiffwriter.js#L320

Reply:
Also, on reviewing the code I write, I can see it's super confusing and it's not clear what type something is.  I'm definitely open to have it being re-written to be more clear as long as people can still pass in a simple array :-)

Reply:
* When it is a normal js array, does it assume that it is 8 bits?

Reply:
@m0ose yes, that is correct.   That is an oversight on my part.  My bad.  Perhaps, you need to write some code to read `metadata.BitsPerSample` and `metadata.SampleFormat` and set what typed array to use based on that?  I believe for Float32, `BitsPerSample` should equal `32` and `SampleFormat` should equal `3`.  @constantinius might have insight of the tiff tags, too.

References:
- https://www.awaresystems.be/imaging/tiff/tifftags/bitspersample.html
- https://www.awaresystems.be/imaging/tiff/tifftags/sampleformat.html